test/density_interface.jl

```diff
@@ -0,0 +1,37 @@
+@testset "DensityInterface" begin
+    using DensityInterface
+
+    d_uv_continous = Normal(-1.5, 2.3)
+    d_uv_discrete = Poisson(4.7)
+    d_mv = MvNormal([2.3 0.4; 0.4 1.2])
+    d_av = Distributions.MatrixReshaped(MvNormal(rand(10)), 2, 5)
+
+    @testset "Distribution" begin
+        for d in (d_uv_continous, d_uv_discrete, d_mv, d_av)
+            x = rand(d)
+            ref_logd_at_x = logpdf(d, x)
+            DensityInterface.test_density_interface(d, x, ref_logd_at_x)
+
+            # Stricter than required by test_density_interface:
+            @test logfuncdensity(logdensityof(d)) === d
```

Comment:
How does this work?

Reply:
It's the default way `DensityInterface.logfuncdensity` behaves, but it's not required for specializations of it. So strictly speaking, we don't need to test it here - I just thought it would be nice to formally require it for distributions.

Reply:
Yes, but isn't `logdensityof(d)` a function? How do you get the distribution back?

Reply:
That's because we have (the equivalent of)

```julia
logdensityof(d) = Base.Fix1(logdensityof, d)

logfuncdensity(log_f::Base.Fix1{typeof(logdensityof)}) = log_f.x
```

in DensityInterface. Makes for a nice symmetry.

Reply:
We can take the `logfuncdensity(logdensityof(d)) === d` for distributions out though, if you prefer. I don't think it's really important.

Reply:
No, it's fine, just wanted to check how it works.

Reply:
Ok, thanks!